picard/tests/test_solver.py

```diff
@@ -206,3 +206,31 @@ def score_and_der(self, Y):
 def test_fun():
     for fun in [Tanh(), Exp(), Cube()]:
         check_density(fun)
+
+
+def test_no_regression():
+    n_tests = 10
+    baseline = {}
+    baseline['lap', True] = 17.
+    baseline['lap', False] = 23.
+    baseline['gauss', True] = 52.
+    baseline['gauss', False] = 60.
+    N, T = 10, 1000
+    for mode in ['lap', 'gauss']:
+        for ortho in [True, False]:
+            n_iters = []
+            for i in range(n_tests):
+                rng = np.random.RandomState(i)
+                if mode == 'lap':
+                    S = rng.laplace(size=(N, T))
+                else:
+                    S = rng.randn(N, T)
+                A = rng.randn(N, N)
+                X = np.dot(A, S)
+                _, _, _, n_iter = picard(X.copy(), return_n_iter=True,
```

Comment:
why X.copy()

picard modifies X in place?

Reply:
Fixed